lib/SILOptimizer/Mandatory/TFUtilities.cpp

```diff
@@ -258,6 +259,27 @@ SILLocation tf::getUserSourceLocation(SILInstruction *inst) {
   return getUserSourceLocation(inst->getDebugLocation());
 }
 
+bool tf::isStatefulOp(const GraphOperationInst *graphOp) {
+  StringRef mangledName = graphOp->getName().str();
+  // Is this a known stateful op used in partitioning?
+  if (mangledName.startswith("tfc.SendToHost") ||
+      mangledName.startswith("tfc.RecvFromHost")) {
+    return true;
+  }
+  // Is this a stateful TensorFlow op?
+  std::unique_ptr<TF_Status, decltype(&TF_DeleteStatus)> status(
+    TF_NewStatus(), TF_DeleteStatus);
+  std::string opName = mangledName.substr(0, mangledName.find(','));
```

Comment:
intead of demangling by hand, could/should we use the graph op decoder (e.g. https://github.com/apple/swift/blob/74cbb5f68d6178a396cc62c8c1f403af5cc49500/lib/SILOptimizer/Mandatory/TFLowerGraph.cpp#L1542)?